### ISSUE TITLE
Security Email subclasses such as `Member_ForgotPasswordEmail` can cause issues with injector overrides.

### ISSUE THREAD

**Issue by bummzack** [silverstripe/silverstripe-framework#5244]
Consider the following case: There's a custom `Email` subclass that uses Emogrifier to inline CSS.
Email is thus overridden via Injector config like this:

```yaml
Injector:
  Email:
    class: 'EmogrifierEmail'
```

Subclasses of `Email` will now also create instances of `EmogrifierEmail`, but they will lose all functionality defined in the subclass. This affects `Member_ChangePasswordEmail` and `Member_ForgotPasswordEmail`.

These two email subclasses seem to be superfluous anyway. One could simply change: 
```php
$e = Member_ForgotPasswordEmail::create();
```
to:
```php
$e = Email::create();
$e->setTemplate('ForgotPasswordEmail');
$e->setSubject(_t('Member.SUBJECTPASSWORDRESET', "Your password reset link", 'Email subject'));
```

Which would result in identical behavior and *not* break injector overrides.

**TL/DR**: Deprecate `Member_ForgotPasswordEmail` and `Member_ChangePasswordEmail` in favor of simply using `Email` and setting template and subject where the email is being sent.

**Comment by tractorcow** [comment]
Hm, I think this is a duplicate issue of a larger problem... that is injector overrides are breaking subclasses of overridden parents. It's not just specific to Email.

I think we need to change Injector to not use inherited configs.

**Comment by tractorcow** [comment]
Duplicate of https://github.com/silverstripe/silverstripe-framework/issues/4774

**Comment by bummzack** [comment]
@tractorcow I agree that this is a duplicate of the other issue. Although I also think it would make sense to deprecate the two Email subclasses in question.

**Comment by tractorcow** [comment]
That would be fine... your suggested improvement seems fine to me. Wan to knock up a PR on master?

**Comment by bummzack** [comment]
@tractorcow Ok, will do. Since this would go into master (4.0), for what version should I mark the classes as deprecated? 5.0?

**Comment by tractorcow** [comment]
If you are pushing to master, you can directly remove them, since you are allowed to break API.

**Comment by bummzack** [comment]
@tractorcow PR is up #5271, I had to alter/fix `TestMailer` though.

**Comment by tractorcow** [comment]
Thanks very much!
